pkg/detector/detector.go

```diff
@@ -359,63 +365,59 @@ func (d *ResourceDetector) LookForMatchedPolicy(object *unstructured.Unstructure
 	}
 
 	klog.V(2).Infof("Attempts to match policy for resource(%s)", objectKey)
-	policyObjects, err := d.propagationPolicyLister.ByNamespace(objectKey.Namespace).List(labels.Everything())
+	policyList := &policyv1alpha1.PropagationPolicyList{}
+	err := d.Client.List(context.TODO(), policyList, &client.ListOptions{
+		Namespace:             objectKey.Namespace,
+		UnsafeDisableDeepCopy: ptr.To(true),
+	})
 	if err != nil {
 		klog.Errorf("Failed to list propagation policy: %v", err)
 		return nil, err
 	}
-	if len(policyObjects) == 0 {
+	if len(policyList.Items) == 0 {
 		klog.V(2).Infof("No propagationpolicy find in namespace(%s).", objectKey.Namespace)
 		return nil, nil
 	}
-
-	policyList := make([]*policyv1alpha1.PropagationPolicy, 0)
-	for index := range policyObjects {
-		policy := &policyv1alpha1.PropagationPolicy{}
-		if err = helper.ConvertToTypedObject(policyObjects[index], policy); err != nil {
-			klog.Errorf("Failed to convert PropagationPolicy from unstructured object: %v", err)
-			return nil, err
-		}
-
+	policies := make([]*policyv1alpha1.PropagationPolicy, 0, len(policyList.Items))
+	for _, policy := range policyList.Items {
 		if !policy.DeletionTimestamp.IsZero() {
 			klog.V(4).Infof("Propagation policy(%s/%s) cannot match any resource template because it's being deleted.", policy.Namespace, policy.Name)
 			continue
 		}
-		policyList = append(policyList, policy)
+		policyPtr := &policy
+		policies = append(policies, policyPtr)
 	}
 
-	return getHighestPriorityPropagationPolicy(policyList, object, objectKey), nil
+	return getHighestPriorityPropagationPolicy(policies, object, objectKey), nil
 }
 
 // LookForMatchedClusterPolicy tries to find a ClusterPropagationPolicy for object referenced by object key.
 func (d *ResourceDetector) LookForMatchedClusterPolicy(object *unstructured.Unstructured, objectKey keys.ClusterWideKey) (*policyv1alpha1.ClusterPropagationPolicy, error) {
 	klog.V(2).Infof("Attempts to match cluster policy for resource(%s)", objectKey)
-	policyObjects, err := d.clusterPropagationPolicyLister.List(labels.Everything())
+	policyList := &policyv1alpha1.ClusterPropagationPolicyList{}
+	err := d.Client.List(context.TODO(), policyList, &client.ListOptions{
+		UnsafeDisableDeepCopy: ptr.To(true),
```

Comment:
would you consider move two of this to https://github.com/karmada-io/karmada/pull/5813? I think this feature needs to be considered as a whole.

Reply:
> I think this feature needs to be considered as a whole.

This is new code, you mean remove UnsafeDisableDeepCopy in this pr, and add in pr 5813?

Reply:
ah, yes.

Reply:
so, This pr must br merged before 5813, but it's not guaranteed

Reply:
After reading your discussion under #5813, I agree that it is better to move this setting to #5813 for unified discussion. 

On the one hand, we can consider the overall performance improvement brought by this setting and on the other hand, we can think about how to better avoid potential program threats brought by this setting, if there is no good protection measure. It seems a little difficult to block threats with reviewers alone.